lib/timeline/component/ItemSet.js

```diff
@@ -681,13 +681,37 @@ ItemSet.prototype.redraw = function() {
   // redraw the background group
   this.groups[BACKGROUND].redraw(range, nonFirstMargin, forceRestack);
 
-  // redraw all regular groups
-  util.forEach(this.groups, function (group) {
-    var groupMargin = (group == firstGroup) ? firstMargin : nonFirstMargin;
-    var groupResized = group.redraw(range, groupMargin, forceRestack);
-    resized = groupResized || resized;
-    height += group.height;
+  var redrawQueue = {};
+  var redrawQueueLength = 0;
+
+  // collect redraw functions
+  util.forEach(this.groups, function (group, key) {
+    if (key === BACKGROUND) return;
+    var groupMargin = group == firstGroup ? firstMargin : nonFirstMargin;
+    var returnQueue = true;
+    redrawQueue[key] = group.redraw(range, groupMargin, forceRestack, returnQueue);
+    redrawQueueLength = redrawQueue[key].length;
   });
+
+  if (redrawQueueLength) {
+    var redrawResults = {};
+
+    for (var i = 0; i < redrawQueueLength; i++) {
+      util.forEach(redrawQueue, function (fns, key) {
+        redrawResults[key] = fns[i]();
```

Comment:
But then the `[i]` does not make sense. Why go through all the trouble of creating a `Queue` instance, when the separate items in it are meant for set individual items?